app/javascript/components/ResetPassword.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import Axios from 'axios';
+
+export default class ResetPasswordForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { user: { password: '',
+                           password_confirmation: '',
+                           "access-token": this.props.location.search.match(/access-token=(.+)&client=/)[1],
```

Comment:
this parsing of params looks a little bit messy, can you parse with some js package?